app/lib/collection-role.coffee

```diff
@@ -0,0 +1,12 @@
+apiClient = require '../api/client'
+
+module.exports =
+  hasSettingsRole: ->
+    if @props.user?
+      apiClient.type('collection_roles').get(collection_id: @props.collection.id, user_id: @props.user.id)
+        .catch ->
+          false
+        .then ([userRoles]) ->
+          userRoles.roles.some (role) -> role.match ///owner|collaborator///i
```

Comment:
In this instance, .then will run every time. catch only catches an error that happens before it in the chain, in this case if the API call itself threw. The chain continues to the next `then` statement.

There isn't any matching per se that's going on here. `.then ([userRoles]) ->` will run everytime, and attempt to destructure the first argument passed back and assign `userRoles` as the first element of the first argument. If you don't have any roles on the collection, it'll be assigned `undefined`, leading to an error when you try to access the `role` property.